scene_switcher.gd

```diff
@@ -1,60 +1,75 @@
 extends ColorRect
+signal scene_changed(scene_name)
 
 @onready var animator = $AnimationPlayer
 
-var next_scene
+#var _start_scene: String
+
+var next_scene_path
 var player_pos
+var left_door_animation
+var right_door_animation 
+var next_scene
 
 const HEAD_OFFSET = Vector3(0, 0.15, 0)
 
-# Called when the node enters the scene tree for the first time.
 func _ready():
 	visible = false
 
 
-# Called every frame. 'delta' is the elapsed time since the previous frame.
-#func _process(delta):
-#	if Input.is_action_just_pressed("jump"):
-#		switch_scene("some")
-
-func fade_to_black():
-	var camera = get_viewport().get_camera_3d()
-	player_pos = camera.unproject_position(camera.target.global_position + HEAD_OFFSET)
-	material.set_shader_parameter("center", player_pos)
-	animator.play("Out")
-	visible = true
-	await animator.animation_finished
-
-func unfade_from_black(delay: float = 0.0):
-	if delay > 0.0:
-		await get_tree().create_timer(delay).timeout
-	var camera = get_viewport().get_camera_3d()
-	player_pos = camera.unproject_position(camera.target.global_position + HEAD_OFFSET)
-	material.set_shader_parameter("center", player_pos)
-	animator.play("In")
-	await animator.animation_finished
-	visible = false
+func _on_scene_changed(scene_path):
+	var scene_name = scene_path.get_file().get_basename()
+	print("Scene name :", scene_name)
+
+#func fade_to_black():
+	#var camera = get_viewport().get_camera_3d()
+	#player_pos = camera.unproject_position(camera.target.global_position + HEAD_OFFSET)
+	#material.set_shader_parameter("center", player_pos)
+	#animator.play("Out")
+	#visible = true
+	#await animator.animation_finished
 
-func switch_scene(scene):
-	next_scene = scene
-	await fade_to_black()
-	var bgm = Global.UI.get_node_or_null("BGM")
-	if bgm:
-		bgm.reparent(self)
-	else:
-		bgm = get_node("BGM")
-	var next: PackedScene = load(next_scene)
+#func unfade_from_black(delay: float = 0.0):
+	#if delay > 0.0:
+	#	await get_tree().create_timer(delay).timeout
+	#var camera = get_viewport().get_camera_3d()
+	#player_pos = camera.unproject_position(camera.target.global_position + HEAD_OFFSET)
+	#material.set_shader_parameter("center", player_pos)
+	#animator.play("In")
+	#await animator.animation_finished
+	#visible = false
+
+func switch_scene(scene_path):
+	next_scene_path = scene_path
+	print("Switching to scene: ", next_scene_path)
+	#await fade_to_black()
+	#var bgm = Global.UI.get_node_or_null("BGM")
```

Comment:
you broke the automatic BGM switching here

Reply:
I assumed it was not implemented at the moment. Alright im going to fix that.